### PR TITLE
Globalsun: Add new bidder

### PR DESCRIPTION
**Pull request by ValeriiBarsuk**
Globalsun: Add new bidder

### CODE REVIEW

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

File: src/main/resources/bidder-config/globalsun.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+adapters:
+  globalsun:
+    endpoint: http://endpoint.globalsun.io/pserver
+    meta-info:
+      maintainer-email: prebid@globalsun.io
+      app-media-types:
+        - banner
+        - video
+        - native
+      site-media-type:
```
Comment (And1sS) [inline src/main/resources/bidder-config/globalsun.yaml:10]: Wrong spelling, it should be `site-media-types`.

**Review by And1sS — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/globalsun/model/GlobalsunImpExtBidder.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import lombok.Builder;
+import lombok.Value;
+
+@Builder
+@Value
+public class GlobalsunImpExtBidder {
+
+    String type;
+
+    @JsonProperty(value = "placementId")
```
[the remark this answers is not in the document]
Comment (SerhiiNahornyi) [reply src/main/java/org/prebid/server/bidder/globalsun/model/GlobalsunImpExtBidder.java:11]: I mean you can write like this
`@JsonProperty("placementId")`

File: src/main/java/org/prebid/server/bidder/globalsun/GlobalsunBidder.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    private ExtImpGlobalsun parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), GLOBALSUN_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
+        }
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpGlobalsun extImpAndGlobalsun) {
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/globalsun/GlobalsunBidder.java:83]: Pls put methods in call order
makeRequests
... makeRequest related
makeBids
...makeBids related

File: src/main/java/org/prebid/server/bidder/globalsun/GlobalsunBidder.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        return HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri(endpointUrl)
+                .headers(HttpUtil.headers())
+                .body(mapper.encodeToBytes(outgoingRequest))
+                .payload(outgoingRequest)
+                .build();
+    }
+
+    private GlobalsunImpExtBidder resolveOutgoingImpExt(ExtImpGlobalsun extImpAndGlobalsun) {
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/globalsun/GlobalsunBidder.java:106]: Can be static

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

**Review by SerhiiNahornyi — APPROVED** [review]

**Review by And1sS — APPROVED** [review]

**Pull request merged**